### DISCUSSION TITLE
Is there overhead / mem wastage to extensive `.val` reads?

### DISCUSSION THREAD

**Discussion by metaleap** [vanjs-org/van#124] [Q&A] [answered]
Trying to grok the `van-1.2.0.js` (usually I'm a backend coder), I noticed the `curDeps?.add(this)` in `get val()` and this got me thinking whether every single innocent "variable-read like" `get` access pushes something onto some `curDeps` list. Or is that one always undefined/null during "userland" read accesses that occur ad-hoc/whenever, rather than inside vanJS' own internal lifecycle functionality?

**Comment by Tao-VanJS** [comment, marked as answer]
`curDeps` is always `undefined` if `val` is being accessed outside binding functions.

Regarding the performance penalty, I have submitted the benchmark of VanJS into https://github.com/krausest/js-framework-benchmark (see https://github.com/krausest/js-framework-benchmark/pull/1353). During the submission, I tried a few micro-optimization techniques, including to avoid capturing the dependencies every time the binding function is called if we can know the dependencies ahead of time. But none of the micro-optimizations lead to statistical difference in the benchmark result. So I guess the overhead of accessing `val` property should be minimal.

That said, if you are accessing `val` property a lot in a binding function, you're encouraged to store the value into a local variable to avoid the repetitive `val` access and perhaps do some benchmark to see if it helps the performance.
